subsys/bluetooth/mesh/vnd/silvair_enocean_srv.c

```diff
@@ -183,14 +188,19 @@ static void timeout(struct k_work *work)
 		button->state = BT_MESH_SILVAIR_ENOCEAN_STATE_DELTA;
 		button->tick_count = 0;
 		send_delta(button, PHASE_B);
+		button->onoff_disabled = true;
 		break;
 	case BT_MESH_SILVAIR_ENOCEAN_STATE_DELTA:
 		send_delta(button, PHASE_C);
-		if (button->tick_count++ <
-		    CONFIG_BT_MESH_SILVAIR_ENOCEAN_DELTA_TIMEOUT /
-		    CONFIG_BT_MESH_SILVAIR_ENOCEAN_TICK_TIME) {
+
+		if ((button->tick_count++ < MAX_TICKS) &&
+		    (k_uptime_get() - button->start_ts <
+		     CONFIG_BT_MESH_SILVAIR_ENOCEAN_DELTA_TIMEOUT)) {
```

Comment:
Could you explain why this check is needed? If I don't miss anything, `CONFIG_BT_MESH_SILVAIR_ENOCEAN_DELTA_TIMEOUT` starts after Phase B (or when button's state is set to `BT_MESH_SILVAIR_ENOCEAN_STATE_DELTA`, not when button is pressed. I though `ticks_count` should cover this case.

Reply:
I guess that the point of this check is to satisfy this spec definition:
![image](https://user-images.githubusercontent.com/35562897/151149610-3170661f-2cc5-4dc0-9947-99ad87b4f9ba.png)

If this assumption is correct it is IMO incorrect to start the timing from any other point than the button press event. I have posted a description of why this is an issue in the JIRA issue, where you can see that it takes approx 7.5 sec before the initial timer scheme triggers.

Reply:
> I guess that the reason for this spec requirement is due to the fact that a button press of 7 sec is associated with the commisioning procedure of the enocean switch.

This is a good point. But then I guess `CONFIG_BT_MESH_SILVAIR_ENOCEAN_DELTA_TIMEOUT ` can't be used in the second expression, otherwise there is no point in the first expression in this if-statement. I guess that `MAX_TICKS` could be defined as:
```
#define MAX_TICKS ((CONFIG_BT_MESH_SILVAIR_ENOCEAN_DELTA_TIMEOUT - CONFIG_BT_MESH_SILVAIR_ENOCEAN_WAIT_TIME) / \
		  CONFIG_BT_MESH_SILVAIR_ENOCEAN_TICK_TIME)
```
and there will be no need in `start_ts`. And definition of `CONFIG_BT_MESH_SILVAIR_ENOCEAN_DELTA_TIMEOUT` should be updated in Kconfig.